plugins/woocommerce/src/Admin/Features/LaunchYourStore.php

```diff
@@ -93,4 +96,38 @@ public function preload_settings( $settings ) {
 
 		return $settings;
 	}
+
+	/**
+	 * Add 'coming soon' banner on the frontend when the following conditions met.
+	 *
+	 * - User must be either an admin or store editor (must be logged in).
+	 * - 'woocommerce_coming_soon' option value must be 'yes'
+	 */
+	public function maybe_add_coming_soon_banner_on_frontend() {
+		// User must be an admin or editor.
+		// phpcs:ignore
+		if ( ! current_user_can( 'editor' ) && ! current_user_can( 'administrator' ) ) {
+			return false;
+		}
+
+		// 'woocommerce_coming_soon' must be 'yes'
+		if ( get_option( 'woocommerce_coming_soon', 'no' ) !== 'yes' ) {
+			return false;
```

Comment:
We also need to check whether it's store pages only. I'm working on a general function for determining if a front-end URL needs a coming soon -- we could follow-up and use that here once it's ready.

Reply:
@adrianduffell Sounds good. What's the condition/requirement for that? Do we use a different copy?

Do we want to merge this and follow-up in a separate PR for that?

Reply:
Yeah we could follow-up in a separate PR. It would have the same copy, there would just need to be a more complex check.

We already have `WCAdminHelper::is_store_page` -- if the coming soon mode is "store pages only", we check the existing URL against `WCAdminHelper::is_store_page` to see if the banner needs to show.